service/src/main/java/greencity/service/CommentServiceImpl.java

```diff
@@ -237,6 +251,11 @@ public PageableDto<CommentDto> getAllActiveComments(Pageable pageable, UserVO us
             if (habit.isEmpty()) {
                 throw new NotFoundException(HABIT_NOT_FOUND_BY_ID + articleId);
             }
+        } else if (articleType == ArticleType.ECO_NEWS) {
+            Optional<EcoNews> ecoNews = ecoNewsRepo.findById(articleId);
+            if (ecoNews.isEmpty()) {
+                throw new NotFoundException(ECO_NEWS_NOT_FOUND_BY_ID + articleId);
+            }
         }
```

Comment:
Why do we need this?
```
if (articleType == ArticleType.HABIT) {
            Optional<Habit> habit = habitRepo.findById(articleId);
            if (habit.isEmpty()) {
                throw new NotFoundException(HABIT_NOT_FOUND_BY_ID + articleId);
            }
        } else if (articleType == ArticleType.ECO_NEWS) {
            Optional<EcoNews> ecoNews = ecoNewsRepo.findById(articleId);
            if (ecoNews.isEmpty()) {
                throw new NotFoundException(ECO_NEWS_NOT_FOUND_BY_ID + articleId);
            }
        }``` ?
`habit` and `ecoNews` are never used

Reply:
As far as I understand, previous author wanted to throw an exception to indicate that we're trying to get comments for habit/econews that doesn't exist.

Reply:
it should be extracted to a separate method to follow the Single responsibility principle